Workbooks/HDInsight/Overview/Overview.workbook

```diff
@@ -0,0 +1,442 @@
+        "style": "pills",
+        "queryType": 0,
+        "resourceType": "microsoft.operationalinsights/workspaces"
+      },
+      "name": "parameters - 2"
+    },
+    {
+      "type": 3,
+      "content": {
+        "version": "KqlItem/1.0",
+        "query": "metrics_kafka_CL\r\n| where InstanceName_s == \"kafka-KafkaController-ActiveControllerCount-Value\" or InstanceName_s == \"kafka-ReplicaManager-LeaderCount-Value\" or InstanceName_s == \"kafka-ReplicaManager-PartitionCount-Value\" or InstanceName_s == \"kafka-ReplicaManager-UnderReplicatedPartitions-Value\" or InstanceName_s == \"kafka-KafkaController-OfflinePartitionsCount-Value\"\r\n| summarize arg_max(TimeGenerated, *) by InstanceName_s, HostName_s\r\n| summarize \r\nAggControllerCount=avg(kafka_KafkaController_ActiveControllerCount_Value_value_d), \r\nAggLeaderCount=avg(kafka_ReplicaManager_LeaderCount_Value_value_d),\r\nAggPartitionCount=avg(kafka_ReplicaManager_PartitionCount_Value_value_d),\r\nAggUnderReplicatedPartitionCount=avg(kafka_ReplicaManager_UnderReplicatedPartitions_Value_value_d),\r\nAggOfflinePartitionCount=avg(kafka_KafkaController_OfflinePartitionsCount_Value_value_d)\r\nby bin(TimeGenerated, 5m), ClusterName_s, HostName_s\r\n| summarize \r\nActiveControllerCount=sum(AggControllerCount),\r\nLeaderCount=sum(AggLeaderCount),\r\nPartitionCount=sum(AggLeaderCount),\r\nUnderReplicatedPartitionCount=sum(AggUnderReplicatedPartitionCount),\r\nOfflinePartitionCount=sum(AggOfflinePartitionCount)\r\nby ClusterName_s\r\n| join kind=leftouter (\r\n   metrics_kafka_CL\r\n    | where InstanceName_s == \"kafka-ReplicaManager-PartitionCount-Value\"\r\n    | distinct HostName_s, ClusterName_s\r\n    | summarize BrokerCount=count() by ClusterName_s\r\n) on ClusterName_s\r\n| extend first6 = substring(ClusterName_s, 0, 6)\r\n| join kind=leftouter (\r\n    Perf\r\n    | where ObjectName == \"Logical Disk\" and (CounterName == \"Free Megabytes\" or CounterName == \"% Free Space\") and InstanceName contains \"data_disk\" and Computer               startswith_cs \"wn\"\r\n    | summarize arg_max(TimeGenerated, *) by Computer, CounterName, InstanceName\r\n    | extend PercentFree=iff(CounterName == \"% Free Space\", CounterValue, real(0)), FreeMegabytes=iff(CounterName == \"Free Megabytes\", CounterValue, real(0))\r\n    | summarize PercentFree=sum(PercentFree), FreeMegabytes=sum(FreeMegabytes) by Computer, InstanceName\r\n    | extend TotalMegabytes=round(FreeMegabytes / PercentFree * 100)\r\n    | extend first6 = trim_start(\".*-\", Computer)\r\n    | summarize FreeMegabytesPerCluster = sum(FreeMegabytes), TotalMegabytesPerCluster = sum(TotalMegabytes) by first6\r\n    | extend DiskPercentFree = round(FreeMegabytesPerCluster/TotalMegabytesPerCluster*100)\r\n) on first6\r\n| project Name=ClusterName_s, ActiveControllerCount, DiskPercentFree, PartitionsPerBroker=round(PartitionCount/BrokerCount), PartitionCount, UnderReplicatedPartitionCount, OfflinePartitionCount, LeaderCount\r\n\r\n",
```

Comment:
if the workspace doesn't have this table at all, this query will generate this error instead:

`'where' operator: Failed to resolve table or column expression named 'metrics_kafka_CL'...`

you might need to do a simple tableifexists query in a hidden parameter, and then use the results of that query to hide these items, and show other text step that says this workspace isn't onboarded/etc?

Reply:
apparently there isn't a tableifexists, just columnifexists.  kustalk had some way of unioning between a datatable inside a function to make it work though?

Reply:
fixed, thanks